src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3572,6 +3572,12 @@ private IOperation VisitNoneOperationExpression(IOperation operation)
             return Operation.CreateOperationNone(semanticModel: null, operation.Syntax, operation.ConstantValue, childrenBuilder.ToImmutableAndFree(), operation.IsImplicit);
         }
 
+        public override IOperation VisitNameOf(INameOfOperation operation, int? captureIdForResult)
+        {
+            Debug.Assert(operation.ConstantValue.HasValue);
```

Comment:
>operation.ConstantValue.HasValue [](start = 25, length = 32)

Is this assert going to hold for error cases?

Reply:
@chborl looks like there's some case that this won't hold for VB: https://github.com/dotnet/roslyn/blob/master/src/Compilers/VisualBasic/Portable/Binding/Binder_Expressions.vb#L692-L694. The field isn't allowed to be null in C#.

Reply:
Will this not have a ConstantValue with `null` as the value, which means the assert still holds?

Reply:
Yes, I verified this with VB test NameOfFlow_01 which is an error case.  ConstantValue = null and HasValue is still true.